src/compiler/checker.ts

```diff
@@ -18659,7 +18659,8 @@ namespace ts {
                 // Before normalization: if `source` is type an object type, and `target` is primitive,
                 // skip all the checks we don't need and just return `isSimpleTypeRelatedTo` result
                 if (originalSource.flags & TypeFlags.Object && originalTarget.flags & TypeFlags.Primitive) {
-                    if (isSimpleTypeRelatedTo(originalSource, originalTarget, relation, reportErrors ? reportError : undefined)) {
+                    if (relation === comparableRelation && !(originalTarget.flags & TypeFlags.Never) && isSimpleTypeRelatedTo(originalTarget, originalSource, relation) ||

```

Comment:
The condition was adjusted based on:
https://github.dev/microsoft/TypeScript/blob/8636adbbb9ab47bb377bd89a3390a13b66df34d2/src/compiler/checker.ts#L18248

and also the same~ thing being checked a few lines below in this very same function:
https://github.dev/microsoft/TypeScript/blob/8636adbbb9ab47bb377bd89a3390a13b66df34d2/src/compiler/checker.ts#L18709-L18710